### PR TITLE
Refactor grandstack bin and add templating from starter

### PR DESCRIPTION
**Pull request by ed42311**
Rewrite and refactor on create-grandstack-app codebase

  - Arg over commander for explicit selection criteria
  - Move configuration into main listr stack, check on packaging options

Updates to functionality

   - [x] Template selection current supports angular, react-ts and react
   - [x] git initialization 
   - [x] setup dev start and build through template scripts

### CODE REVIEW

File: src/utils/main.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            {
+              title: `Installing ${templateName} dependencies`,
+              task: () =>
+                projectInstall({
+                  cwd: path.join(newAppDir, templateName),
+                  prefer: packageManager,
+                }),
+            },
+          ]),
+        skip: () =>
```
Comment (johnymontana) [inline src/utils/main.js:72]: I think dependencies should be installed by default

File: src/utils/options.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import arg from 'arg'
+import chalk from 'chalk'
+
+const templateOpts = ['React', 'React-TS', 'Angular']
+const APIQuestions = [
+  {
+    type: 'input',
+    name: 'whatever',
+    message: chalk.green(
+      `Now let's configure your GraphQL API to connect to Neo4j. Take a minute to create a graph instance.\nAll Set?\n`
```
Comment (johnymontana) [inline src/utils/options.js:11]: Change to "Now let's configure your GraphQL API to connect to Neo4j. If you don't have a Neo4j instance you can create one for free in the cloud at https://neo4j.com/sandbox"

File: src/utils/file.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  const result = await execa('git', ['init'], {
+    cwd: newAppDir,
+  })
+  if (result.failed) {
+    return Promise.reject(new Error('Failed to initialize git'))
+  }
+  return
+}
+
+export const checkAppDir = (targetDir) => {
```
Comment (johnymontana) [inline src/utils/file.js:23]: The check for if the directory already exists should happen before prompting the user for connection credentials / template. The user will be frustrated to go through the configuration process and then be told they can't create the project.

File: src/utils/options.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      choices: templateOpts,
+      default: defaultTemplate,
+    })
+  }
+
+  if (!runInstall) {
+    questions.push({
+      type: 'confirm',
+      name: 'runInstall',
+      message: 'Install dependencies in all directories?',
```
Comment (johnymontana) [inline src/utils/options.js:108]: Remove "in all directories" and make default = true

**Comment by johnymontana** [comment]
We need some way to tell the `scripts/start-dev.js` script in the grand-stack-starter project to know which project to run. For example, if you choose any template other than react then running `npm run start` after install will fail because `start-dev.js` currently only starts `web-react`

File: src/utils/file.js (hunk 3 above)
Comment (ed42311) [reply src/utils/file.js:23]: Screenshot from 2020-08-14 11-00-50
dies after first input if file exists and is not empty

**Pull request merged**